### ISSUE TITLE
Remove Telerik

### ISSUE THREAD

**Issue by MaiklT** [DNNCommunity/DNN.Faq#59]
Not a bug, but the Telerik dependencies have to be removed.

## To Reproduce
1. Install the [Telerik Identifier Module](https://github.com/IowaComputerGurus/DnnTelerikIdentifier) on a site where DNN.FAQ is installed
2. See messages from this module

## Expected behavior
No Telerik dependency

**Comment by valadas** [comment]
Yep, this is on my todo list, I should be doing this in a couple of weeks...

**Comment by JRPonte** [comment]
Did you look at this? When can be expected a new version?

**Comment by valadas** [comment]
I took a look at it last week and there is way more Telerik usage then I thought initially, it's still my next target but will need some time.

**Comment by brewadmin** [comment]
The Telerik dependency is a concern - and we have many websites using this module

**Comment by valadas** [comment]
I may be rebuilding this module from scratch in a couple of months, but if any PR comes in by then that removes all or parts of Telerik dependencies, any help is appreciated.

**Comment by valadas** [comment]
This module controls almost all depend on Telerik, this might actually end up in a total rebuild just keeping the same data and api but a totally new UI. It would still upgrade from earlier versions though. But it's a larger task than anticipated on such a simple module 😄

**Comment by JRPonte** [comment]
Any news on this upgrade? Have you thought of a date when it is expected to have a new release?

**Comment by jncraig** [comment]
Not a solution but some possible alternatives:

1. Sponsor someone to do the work.
2. Find an alternative (2sxc includes a FAQ app)
3. Look for a commercial alternative, one that doesn't include Telerik components.

#1 will earn you karma points in the DNN Community.  #2 will get you a quality open source solution.  #3 will have you moving forward

**Comment by brentil** [comment]
Any progress on this process now that the Telerik is officially listed on the security page as a Critical vulnerability?

**Comment by WillStrohl** [comment]
This is an open-source module, whose updates are done 100% on a volunteer basis.  As Joe & Daniel previously pointed out, there are a number of ways to help get this issue resolved.  This specific issue is just larger and requires a significant amount of effort.

**Comment by brentil** [comment]
I understand that, but if no one is actually putting that effort forward and running the old version is requiring running known critical vulnerabilities, then maybe these extensions should be considered being end of lifed.  If they were EOL it would send a clear signal to not use these anymore because of security reasons and lack of support.

**Comment by valadas** [comment]
This is still on my todo list, I just can't promise when, I need to set aside 1/2 days to rework this, the whole concept of the module is very Telerik centric so the fix is almost to rewrite the whole thing just keeping the data layer in place for upgrades.

**Comment by WillStrohl** [comment]
@brentil  End of life would mean that no one is participating in and updating the project in any way and it has no "owners" or leaders.  That's simply not the case here.  We're talking bout a single (major) issue.  EOL for this project is hardly a valid next step.  :)

**Comment by swade61970** [comment]
Has anyone found any alternatives (other than 2sxc) for converting their FAQs to another paid DNN module?

**Comment by Timo-Breumelhof** [comment]
I use Open Content

**Comment by WillStrohl** [comment]
> I use Open Content

Same.

**Comment by MaiklT** [comment]
Is there an OpenContent Template for FAQ? Could you share please?

**Comment by WillStrohl** [comment]
I'm sure there is, but all you really need to do is copy the HTML structure from any existing desired FAQ example and then use the form builder to duplicate the fields you wish for it to have.  It's super-easy.  

An example of this is the bootstrap accordion:  http://www.openextensions.net/dnn-modules/opencontent/bootstrap

**Comment by JRPonte** [comment]
Any news on this upgrade? It's the only thing (besides events module ... ) that's preventing the upgrade for DNN 9.10 version and the full removal of telerik. It would be a great xmas present for all community.

**Comment by valadas** [comment]
I did take a good look at trying to do this, but the amount of work is way bigger than I initially thought. So I would not hold my breath on this one as it may be quicker to simply migrate to another multi-purpose module 10 times than to revamp this module.

**Comment by EPTamminga** [comment]
The code of the Events module has more than 15 years history. Telerik functions have been build into the module in a number of places at the start of the free Telerik DNN version. Extended Telerik UI functionality has been used. I do not see it being easily replaced by none telerik UI components...

**Comment by erw13n** [comment]
I had successfully remove Telerik from DNN.Faq
Replace the TreeView with Asp.Net Treeview, and some element that depend on it.
There still some minor issue, such as no longer support Drag n Drop, and Highlight selected node.

Updated source code available on https://github.com/DnnFree/DNN.Faq

**Comment by valadas** [comment]
@erw13n can you send a pull request please.
